### DISCUSSION TITLE
Repeatedly displaying and hiding the popup at short intervals does not display the Popup as intended on iOS.

### DISCUSSION THREAD

**Discussion by cat0363** [CommunityToolkit/Maui#1202] [Q&A]
I am using Popup to display the indicator. Indicators using Popup are displayed before the 
start of processing and closed after the end of processing. The length of the processing varies, 
some long, some short.

The Indicator using the Popup is displayed as follows.

    private void ShowIndicator()
    {
        if (pIndicator == null)
        {
            pIndicator = new Indicator();
            this.ShowPopup(pIndicator);
        }
    }

The Indicator using the Popup is closed as follows.

    private void CloseIndicator()
    {
        if (pIndicator != null)
        {
            pIndicator.Close();
            pIndicator = null;
        }
    }

pIndicator is an instance of Indicator class that inherits from Popup class and is a class variable.
Popups are managed by class variables, and only one Popup is displayed at a time.

    private Indicator pIndicator = null;

I use it as follows.

    this.ShowIndicator();
    await Task.Delay(3000);    // <= do the process
    this.CloseIndicator();

If there is no problem, it will be displayed as follows.

![test_001](https://github.com/CommunityToolkit/Maui/assets/125236133/c3282ecb-9e4d-4092-88a0-df0cd495a3e2)

If the following warning is displayed, it will be displayed as follows.

Warning: Attempt to dismiss from view controller <CommunityToolkit_Maui_Core_Views_MauiPopup: 0x16c8d4250> while a presentation or dismiss is in progress!

![test_002](https://github.com/CommunityToolkit/Maui/assets/125236133/fbdc0f9c-31cd-4698-8f67-6edb1613ec48)

A Indicator using an Popup will appear in a position other than the center of the screen and will not close at all.
The above phenomenon rarely occurs.

Below is the code to reproduce the issue.
https://github.com/cat0363/MauiComm-IssuePopup.git

Run the above app on iOS and tap multiple buttons on the screen at the same time.
Originally, when you press the button, heavy processing is executed, and it takes a load on the terminal.
Since heavy processing cannot be reproduced here, press the button repeatedly.

As far as I can see from the warnings, it looks like the object was not destroyed immediately after 
calling the Popup's Close method and is still there.

Am I using Popup wrong? Should I wait for the popup to be completely destroyed after it's closed?
Or is it a Popup bug? Please let me know if you have any good ideas to solve this problem. Thank you.

Additional Information:
I ran it in the following environment.
- .NET MAUI CommunityToolkit:5.1.0
- OS:iOS 16.4
- .NET MAUI:7.0.86

**Comment by cat0363** [comment]
The reproduction code is for reproducing the problem simply, and is different from the situation in which the problem occurred.
In fact, this problem occurs when showing an indicator when additionally retrieving data depending on the ScrollView's scroll position.
If the user scrolls quickly, ShowPopup and Close will be called in quick succession, which causes problems.
If it is difficult to reproduce the problem with the reproduction code, I will think of another reproduction code separately, so please reply.

**Comment by bijington** [comment]
@cat0363 Is it possible to move this over to an issue rather than a discussion? 

I can't tell you what the issue is and whether it is something we have control over but it certainly sounds like we can do something even if it is documenting the use around popups better. It does feel like this should be solvable though

**Reply by cat0363** [reply]
Hi, @bijington 
Thank you for your reply. I'm glad you replied.
I was in trouble for a long time.
I would like to transfer from Discussion to Issue.
If possible, I will write a reproducible code that is easier to reproduce, so please wait for a while.

**Reply by cat0363** [reply]
Hi, @bijington 
Opened Issue #1213.
I'm sorry for your inconvenience.
Thank you for your cooperation.

**Reply by bijington** [reply]
That's great thank you. I am in and working around popups at the moment so I will try and see I can work out what is going on.

**Comment by CliffAgius** [comment]
Closed as moved to issue #1213
